packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx

```diff
@@ -0,0 +1,279 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import * as KeyCode from 'keycode-js';
+import '../_elementPolyfill';
+import WorklistDataGridUtils from '../utils/WorklistDataGridUtils';
+
+const propTypes = {
+  /**
+   * String identifier of the row in which the Cell will be rendered.
+   */
+  rowId: PropTypes.string.isRequired,
+
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+
+  coordinates: PropTypes.shape({
+    x: PropTypes.number,
+    y: PropTypes.number,
+  }),
+
+  /**
+   * Boolean value to indicate if the cell can accept focus.
+   */
+  acceptsFocus: PropTypes.bool,
+
+  /**
+   * Boolean indicating whether the Cell is actively selected.
+   */
+  isSelected: PropTypes.bool,
+
+  /**
+   * @private
+   * Boolean indicating whether the cell is a row selection cell.
+   */
+  isRowSelectionCell: PropTypes.bool,
+
+  /**
+   * String that labels the cell for accessibility.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * Class name for the cell.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Function that is called when a selectable cell is selected. Parameters: `onCellSelect(rowId, columnId)`
+   */
+  onCellSelect: PropTypes.func,
+
+  onCellSelectionChange: PropTypes.func,
+  onMoveCellFocus: PropTypes.func,
+  onNavigationModeChange: PropTypes.func,
+
+  isRowSelectionModeEnabled: PropTypes.bool,
+  isRowHeader: PropTypes.bool,
+  isNavigationEnabled: PropTypes.bool,
+
+  /**
+   * Content that will rendered within the Cell.
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  acceptsFocus: false,
+  isRowSelectionCell: false,
+  isRowHeader: false,
+};
+
+function Cell(props) {
+  const {
+    rowId,
+    columnId,
+    coordinates,
+    acceptsFocus,
+    isSelected,
+    isRowSelectionCell,
+    ariaLabel,
+    className,
+    onCellSelectionChange,
+    onCellSelect,
+    onMoveCellFocus,
+    isRowSelectionModeEnabled,
+    onNavigationModeChange,
+    isRowHeader,
+    children,
+    isNavigationEnabled,
+    ...customProps
+  } = props;
+
+  const selectCell = (event) => {
+    // If current cell is selected, do nothing.
+    if (isSelected) {
+      event.preventDefault();
+      return;
+    }
```

Comment:
The cell probably should not care if it is selected.  It should just return that it was clicked or space was pressed.